### DISCUSSION TITLE
Feat: Organize chats into folders or Organize chats per model used

### DISCUSSION THREAD

**Discussion by quindarius-gooch** [open-webui/open-webui#3158] [Ideas]
**Is your feature request related to a problem? Please describe:** I get lost trying to navigate my previous chats.

**Describe the solution you'd like:**
I use openwebui a lot, it's great! But I end up creating a lot of chats and some of those I reference back to at some point. It would be nice to better group previous chats, whether it was a folder I could add chats to or better yet save chats pertaining to a certain model used. For example I have a model dedicated for assisting me with help desk tickets, every time i selected this model, it would be awesome to see just chats for that model be listed on the sidebar.

**Comment by NCMcClure** [comment]
Completely agree with being able to organize chats into folders. After using Msty and big-AGI, I've found this feature to be immensely helpful for keeping things tidy and easy to find later, but I just wish it was in OpenWebUI!

**Reply by kasxi** [reply]
Agree folders are a game changer for managing chats.

**Comment by rztaylor** [comment]
I've been using tags to try to organize my chats, but they don't work very well for the following reasons:

- They only act as filters. The default view of "all"  doesn't do anything to declutter the chat list.
- The interface for tagging requires multiple clicks and is slow to use (click on menu > click on add tag > select or type a new tag > click save or press Enter)
- There's no way to see which chats aren't tagged, which makes it difficult to keep all chats organized.

Folders on the other hand would:

- Reduce the clutter on the chat list.
- Instantly see which chats are not in a folder (i.e. are at the root level)
- Drag ''n' drop into a folder would be much easier and quicker to use.
- Sub-folders would permit even better organization

**Comment by thiswillbeyourgithub** [comment]
For me, I've always come to the conclusion that the end all be all of stuff management is to have what anki has :
* both a unique hierarchical folder structure (1 chat is in a single folder only, but that folder can itself be in a folder)
* AND hierarchical tags (each chat can have as many hierarchical tags as needed) for example `work::companyA::for_larry computerscience::ui_ux::design has_heureka_moment`.

This is non invasive and flexible to adapt to any workflow in my experience.

Of course entering a tag needs to autocomplete based on any part of the tag and we need a proper chat browser to allow us to properly filter by tags and search for things instead of just sidebar titles. Bulk management too (delete, add/remove tags, change folder)

This will all be needed eventually because we will for sure end up with the RAG feature being able to ingest from our own chats based on tags.

Also creating a new chat should by default/optout prefill the tag fields of the new chat with the values from the previous chat, lowering the amount of things the user has to enter.

I think this should definitely be part of the roadmap. Has any of the devs some thoughts about this? I know it's a bit ambitious but actually nothing from scratch, it's afaik an already battle tested UX setup.

**Comment by red-co** [comment]
+100, I wanted folder structure, I looked at a lot of front ends and found that none of them had folder structure except stupid tavern.

**Comment by dwrout** [comment]
### UI Chat TAB Folder Structure for Organizing Chat Sessions

Here is what I would like to see. Currently, the Chat TAB down the left side of the OpenWebUI interface allows users to manage and view chat sessions in chronological order. However, as the number of sessions grows, it becomes challenging to keep them organized efficiently. To address this, I propose adding a folder structure that enables users to categorize their chat sessions more effectively.

**Rationale:**

- **Improved Organization:** Users will be able to group related chat sessions together, making it easier to find and manage conversations and projects.
- **Enhanced Usability:** A well-organized chat interface enhances the user experience by reducing clutter and providing a clear structure.
- **Better Collaboration:** Team members can collaborate more effectively if they can easily access and share relevant chat sessions based on predefined categories.

**Proposed Features:**

1. **Folder Creation:** Users should be able to create and name folders within the Chat TAB.
2. **Drag-and-Drop Functionality:** Allow users to drag and drop chat sessions into different folders for easy reorganization.
3. **Nested Folders:** Support for creating subfolders to allow for a more hierarchical organization of chat sessions.
4. **Search Within Folders:** Enhance the search functionality to include searching within specific folders.
5. **Default Folder:** Provide an option to set a default folder where new chat sessions are automatically stored until moved.
6. **Permission Settings:** For team environments, allow administrators to set permissions on who can create, move, or delete folders and chat sessions within those folders.
7. **Visual Indicators:** Use visual cues (e.g., color coding) to differentiate between folders and individual chat sessions for quick recognition.

**Expected Benefits:**

- Increased productivity by making it easier to find and manage chat sessions.
- Better collaboration among team members through organized communication channels.
- Enhanced user satisfaction with a more intuitive and customizable interface.

**Comment by 9Somboon** [comment]
It would be good to separate them by model, so that in each model, what the user has chatted about, like [LobeChat](https://github.com/lobehub/lobe-chat) does. I think it's very convenient and good for organizing.

**Comment by Sahas-Ananth** [comment]
I think it would be really useful if we had a 'Create New Chat Within This Folder' button for each folder. That way, the chats can always be organized.

**Comment by fatihky** [comment]
I was about to create a new discussion about subfolders. This would be really helpful feature to have. I have many chats and need sub folders to organize them.

**Comment by horizzons** [comment]
100+ Subfolders are super important. Open WebUI becomes an organizational mess without it.

**Comment by silenceroom** [comment]
I see that you can create a folder B, and then move it into another folder A which will make B as the subfolder of A. 
But didn't find a way to make it work similar to "Multl agent", you can use a # to associate the folder, but the context did not really attach to the chat. 

Wondering if anyone have a better use case of fodler/subfolder?
